pkg/virt-controller/watch/vmi.go

```diff
@@ -830,6 +839,55 @@ func (c *VMIController) allPodsDeleted(vmi *virtv1.VirtualMachineInstance) (bool
 
 }
 
+func (c *VMIController) clearPBDLabelOnCompletedPods(vmi *virtv1.VirtualMachineInstance) error {
+	pods, err := c.listPodsFromNamespace(vmi.Namespace)
+	if err != nil {
+		return err
+	}
+
+	for _, pod := range pods {
+		if pod.DeletionTimestamp != nil {
+			continue
+		} else if !controller.IsControlledBy(pod, vmi) {
+			continue
+		} else if !podIsDown(pod) {
```

Comment:
podIsDown also includes a failed phase, so this will work not only on the `Completed` pod , or am I wrong?

Reply:
the label is removed when a pod is in a finalized state. So, when the phase is Succeeded or Failed.